bokehjs/src/coffee/models/annotations/color_bar.coffee

```diff
@@ -0,0 +1,401 @@
+    if not @visuals.major_tick_line.doit
+      return
+
+    geom = @compute_legend_bbox()
+    coords = @mget('tick_coords').major
+
+    [sx, sy] = [coords[0], coords[1]]
+    [nx, ny] = @mget('normals')
+
+    tin = @mget('major_tick_in')
+    tout = @mget('major_tick_out')
+
+    [x_offset, y_offset] = [geom.image_sx + geom.image_width * nx, geom.image_sy + geom.image_height * ny]
+
+    ctx.save()
+    ctx.translate(x_offset, y_offset)
+    @visuals.major_tick_line.set_value(ctx)
+    for i in [0...sx.length]
+      ctx.beginPath()
+      ctx.moveTo(Math.round(sx[i]+nx*tout), Math.round(sy[i]+ny*tout))
+      ctx.lineTo(Math.round(sx[i]-nx*tin), Math.round(sy[i]-ny*tin))
+      ctx.stroke()
+    ctx.restore()
+
+  _draw_minor_ticks: (ctx) ->
+    if not @visuals.minor_tick_line.doit
+      return
+
+    geom = @compute_legend_bbox()
+    coords = @mget('tick_coords').minor
+
+    [sx, sy] = [coords[0], coords[1]]
+    [nx, ny] = @mget('normals')
+
+    tin = @mget('minor_tick_in')
+    tout = @mget('minor_tick_out')
+
+    [x_offset, y_offset] = [geom.image_sx + geom.image_width * nx, geom.image_sy + geom.image_height * ny]
+
+    ctx.save()
+    ctx.translate(x_offset, y_offset)
+    @visuals.minor_tick_line.set_value(ctx)
+    for i in [0...sx.length]
+      ctx.beginPath()
+      ctx.moveTo(Math.round(sx[i]+nx*tout), Math.round(sy[i]+ny*tout))
+      ctx.lineTo(Math.round(sx[i]-nx*tin), Math.round(sy[i]-ny*tin))
+      ctx.stroke()
+    ctx.restore()
+
+  _draw_major_labels: (ctx) ->
+    if not @visuals.major_label_text.doit
+      return
+
+    geom = @compute_legend_bbox()
+    coords = @mget('tick_coords').major
+
+    [sx, sy] = [coords[0], coords[1]]
+    [nx, ny] = @mget('normals')
+
+    [x_offset, y_offset] = [geom.image_sx + geom.image_width * nx, geom.image_sy + geom.image_height * ny]
+
+    standoff = (@mget('label_standoff') + @mget('major_tick_out'))
+    [x_standoff, y_standoff] = [standoff*nx, standoff*ny]
+
+    switch @mget('orientation')
+      when "vertical" then labels = coords[1]
+      when "horizontal" then labels = coords[0]
+    labels = @mget('formatter').doFormat(labels)
+
+    @visuals.major_label_text.set_value(ctx)
+
+    ctx.save()
+    ctx.translate(x_offset + x_standoff, y_offset + y_standoff)
+    for i in [0...sx.length]
+      ctx.fillText(labels[i],
+                   Math.round(sx[i]+nx*@mget('label_standoff')),
+                   Math.round(sy[i]+ny*@mget('label_standoff')))
+    ctx.restore()
+
+  _draw_title: (ctx) ->
+    if not @visuals.title_text.doit
+      return
+
+    geom = @compute_legend_bbox()
+    @visuals.title_text.set_value(ctx)
+    ctx.fillText(@mget('title'), geom.image_sx, geom.image_sy)
+
+class ColorBar extends Annotation.Model
+  default_view: ColorBarView
+
+  type: 'ColorBar'
+
+  @mixins ['text:major_label_',
+           'text:title_',
+           'line:major_tick_',
+           'line:minor_tick_',
+           'line:border_',
+           'fill:background_',
+  ]
```

Comment:
This style is inconsistent with others. Use newline after `[` and then have one extra level of indentation per line.